src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp

```diff
@@ -1088,6 +1088,29 @@ bool WholeBodyDynamicsDevice::readFTSensors(bool verbose)
             yWarning() << "wholeBodyDynamics warning : FT sensor " << sensorName << " was not readed correctly, using old measurement";
         }
 
+        if (ftMeasurement.size() != 6)
+        {
+            std::string sensorName = estimator.sensors().getSensor(iDynTree::SIX_AXIS_FORCE_TORQUE,ft)->getName();
+            yError() << "wholeBodyDynamics : FT sensor " << sensorName << " has the wrong size of " << ftMeasurement.size() << " instead of 6, returning error.";
+            return false;
+        }
+
+        bool isNaN = false;
+        for (int i=0; i < ftMeasurement.size(); i++)
+        {
+            if( std::isnan(ftMeasurement[i]) )
+            {
+                isNaN = true;
+                break;
+            }
+        }
+        if( isNaN )
```

Comment:
I think this check should remain, sanity checking on data read from another software module is a good practice.